### DISCUSSION TITLE
Question about roi_head/num_bg_samples

### DISCUSSION THREAD

**Discussion by amida47** [facebookresearch/detectron2#4199] [Q&A] [answered]
in the logged losses we have a loss called roi_head/num_bg_samples I am wondering if it represents the number of background samples per image or per batch 

because I am doubting that MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE sets the number of samples for a batch and not a image

example 
I set 
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 10
cfg.SOLVER.IMS_PER_BATCH = 32

then on tensorboard 
![image](https://user-images.githubusercontent.com/49212088/164789403-e99212c0-57cd-48a1-bd9e-0571d3661598.png)

if we agree that these values are logged every batch, then
as you can see if we sum num_bg_samples and num_fg_samples , we will find  a value equal to ROI_HEADS.BATCH_SIZE_PER_IMAGE

**Comment by ppwwyyxx** [comment, marked as answer]
>  I am wondering if it represents the number of background samples per image or per batch

They represent the average number of fg/bg samples per image.
